test/views/UploadAnotherSupportingDocumentViewSpec.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views
+
+import forms.UploadAnotherSupportingDocumentFormProvider
+import models.{DraftAttachment, DraftId, NormalMode, UploadedFile}
+import play.twirl.api.HtmlFormat
+import views.behaviours.ViewBehaviours
+import views.html.UploadAnotherSupportingDocumentView
+
+import java.time.Instant
+
+class UploadAnotherSupportingDocumentViewSpec extends ViewBehaviours {
+
+  val messageKeyPrefix = "uploadAnotherSupportingDocument"
+
+  val form: UploadAnotherSupportingDocumentFormProvider =
+    app.injector.instanceOf[UploadAnotherSupportingDocumentFormProvider]
+
+  val successfulFile: UploadedFile.Success = UploadedFile.Success(
+    reference = "reference",
+    downloadUrl = "downloadUrl",
+    uploadDetails = UploadedFile.UploadDetails(
+      fileName = "fileName",
+      fileMimeType = "fileMimeType",
+      uploadTimestamp = Instant.now(),
+      checksum = "checksum",
+      size = 1
+    )
+  )
+
+  val singleAttachment: Seq[DraftAttachment] = Seq(
+    DraftAttachment(successfulFile, isThisFileConfidential = Some(false))
+  )
+
+  val attachments: Seq[DraftAttachment] = Seq(
+    DraftAttachment(successfulFile, isThisFileConfidential = Some(false)),
+    DraftAttachment(successfulFile, isThisFileConfidential = Some(false))
+  )
+
+  val maxAttachments: Seq[DraftAttachment] = Seq(
+    DraftAttachment(successfulFile, isThisFileConfidential = Some(false)),
+    DraftAttachment(successfulFile, isThisFileConfidential = Some(false)),
+    DraftAttachment(successfulFile, isThisFileConfidential = Some(false)),
+    DraftAttachment(successfulFile, isThisFileConfidential = Some(false)),
+    DraftAttachment(successfulFile, isThisFileConfidential = Some(false))
+  )
```

Comment:
All these and the one below on line 107 can be cleaned up to a method. You can fill when necessary to the number of docs and include a default if you want.  For example:

```
  private def attachments(numOfDocs: Int = 1): Seq[DraftAttachment] = Seq.fill(numOfDocs)(
    DraftAttachment(successfulFile, isThisFileConfidential = Some(false))
  )
```